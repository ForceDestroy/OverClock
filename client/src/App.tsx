import { BrowserRouter, Route, Routes, useLocation } from 'react-router-dom';
import { Box, Grommet } from 'grommet';
import './App.css';

import Home from './pages/Home';
import NotFound from './pages/NotFound';
import Login from './pages/Login';
import EmployeeDashboard from './pages/EmployeeDashboard';
import EmployerDashboard from './pages/EmployerDashboard';
import NavigationTopBar from './components/NavigationTopBar';
import ProfilePage from './pages/ProfilePage';
import Timesheets from './pages/Timesheets';
import Schedule from './pages/Schedule';
import TimeOff from './pages/TimeOff';
import EmployerTimesheets from './pages/EmployerTimesheets';
import Payslip from './pages/Payslip';
import EmployerRequests from './pages/EmployerRequests';
import CompanyPolicy from './pages/CompanyPolicy';

export function App() {
  const { pathname } = useLocation();
  const toggleNav = pathname === '/Login' || pathname === '/';
  const availableRoutes = [
    '/employeedashboard',
    '/employeraccount',
    '/employerdashboard',
    '/employer/timesheets',
    '/employer/requests',
    '/application',
    '/applicationreferral',
    '/applicationmanagement',
    '/companydashboard',
    '/employeedashboard/companypolicy',
    '/profilepage',
    '/employeedashboard/timesheets',
    '/employeedashboard/schedule',
    '/employeedashboard/payslip',
    '/employeedashboard/timeoff',
  ];
  const errorRoute = availableRoutes.includes(pathname.toLowerCase());
  return (
    <div className="App">
      <Grommet full>
        {!toggleNav && errorRoute && <NavigationTopBar />}
        <Box direction="row" height={{ min: '90vh' }} background="light-1">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/Login" element={<Login />} />
            <Route path="/EmployeeDashboard" element={<EmployeeDashboard />} />
            <Route path="/EmployerDashboard" element={<EmployerDashboard />} />
            <Route path="/Employer/Requests" element={<EmployerRequests />} />
            <Route
              path="/Employer/Timesheets"
              element={<EmployerTimesheets />}
            />
            <Route
              path="/EmployeeDashboard/CompanyPolicy"
              element={<CompanyPolicy />}
            />
            <Route path="/ProfilePage" element={<ProfilePage />} />
            <Route
              path="/EmployeeDashboard/Timesheets"
              element={<Timesheets />}
            />
            <Route path="/EmployeeDashboard/Schedule" element={<Schedule />} />
            <Route path="/EmployeeDashboard/Payslip" element={<Payslip />} />
            <Route path="/EmployeeDashboard/TimeOff" element={<TimeOff />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </Box>
      </Grommet>
    </div>
  );
}

export function WrappedApp() {
  return (
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
}
