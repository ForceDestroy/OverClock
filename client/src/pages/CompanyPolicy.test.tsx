import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, vi } from 'vitest';

import axios from 'axios';
import AuthTokenHelper from '../helpers/AuthTokenHelper';
import { App } from '../App';

vi.mock('axios');
vi.mock('../helpers/AuthTokenHelper');

describe('CompanyPolicy', () => {
  afterEach(() => {
    vi.restoreAllMocks();
    vi.clearAllMocks();
  });
  it('Renders Company Policies to match snapshot', () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    // Arrange
    const snap = render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders the company policies page with all the policies', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(screen.queryByText('Company Policy')).toBeInTheDocument();
    expect(screen.queryByText('Health and Safety')).toBeInTheDocument();
    expect(screen.queryByText('Employee Code of Conduct')).toBeInTheDocument();
    expect(screen.queryByText('Leave of Absence')).toBeInTheDocument();
    expect(screen.queryByText('Attendance')).toBeInTheDocument();
    expect(screen.queryByText('Substance Abuse')).toBeInTheDocument();
    expect(screen.queryByText('Schedule')).toBeInTheDocument();
    expect(screen.queryByText('Equal Opportunity')).toBeInTheDocument();
    expect(screen.queryByText('Contact and Address')).toBeInTheDocument();
    expect(screen.queryByText('About the Company')).toBeInTheDocument();
  });
  it('Renders the company policies for health and safety', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const healthBox = screen.getByTestId('healthBox');
    await user.click(healthBox);

    // Assert
    await waitFor(() => {
      expect(
        screen.queryByText('Health and Safety Policies')
      ).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('Health and Safety Policies')
      ).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for employee code of conduct', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const employeeBox = screen.getByTestId('employeeBox');
    await user.click(employeeBox);

    // Assert
    await waitFor(() => {
      expect(
        screen.queryByText('Employee Code of Conduct Policies')
      ).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('Employee Code of Conduct Policies')
      ).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for leave of absence', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const leaveBox = screen.getByTestId('leaveBox');
    await user.click(leaveBox);

    // Assert
    await waitFor(() => {
      expect(
        screen.queryByText('Leave of Absence Policies')
      ).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('Leave of Absence Policies')
      ).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for attendance', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const attendanceBox = screen.getByTestId('attendanceBox');
    await user.click(attendanceBox);

    // Assert
    await waitFor(() => {
      expect(screen.queryByText('Attendance Policies')).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(screen.queryByText('Attendance Policies')).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for substance abuse', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const drugBox = screen.getByTestId('drugBox');
    await user.click(drugBox);

    // Assert
    await waitFor(() => {
      expect(
        screen.queryByText('Substance Abuse Policies')
      ).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('Substance Abuse Policies')
      ).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for schedule', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const scheduleBox = screen.getByTestId('scheduleBox');
    await user.click(scheduleBox);

    // Assert
    await waitFor(() => {
      expect(screen.queryByText('Schedule Policies')).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(screen.queryByText('Schedule Policies')).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for equal opportunity', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const equalBox = screen.getByTestId('equalBox');
    await user.click(equalBox);

    // Assert
    await waitFor(() => {
      expect(
        screen.queryByText('Equal Opportunity Policies')
      ).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('Equal Opportunity Policies')
      ).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for contact and address', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const contactBox = screen.getByTestId('contactBox');
    await user.click(contactBox);

    // Assert
    await waitFor(() => {
      expect(
        screen.queryByText('Contact and Address Info')
      ).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('Contact and Address Info')
      ).not.toBeInTheDocument();
    });
  });
  it('Renders the company policies for about the company', async () => {
    // Mock
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecureToken1234567890'
    );

    (axios.request as jest.Mock).mockReturnValue({
      data: {
        name: 'web test',
      },
      status: 200,
    });

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/EmployeeDashboard/CompanyPolicy']}>
        <App />
      </MemoryRouter>
    );
    const aboutBox = screen.getByTestId('aboutBox');
    await user.click(aboutBox);

    // Assert
    await waitFor(() => {
      expect(screen.queryByText('About the Company Info')).toBeInTheDocument();
    });

    // Cleanup
    const closeButton = screen.getByTestId('closeLayerButton');
    await user.click(closeButton);

    await waitFor(() => {
      expect(
        screen.queryByText('About the Company Info')
      ).not.toBeInTheDocument();
    });
  });
});
