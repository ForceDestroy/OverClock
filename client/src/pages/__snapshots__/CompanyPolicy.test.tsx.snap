// Vitest Snapshot v1

exports[`CompanyPolicy > Renders Company Policies to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
        >
          <div>
            <div
              class="StyledGrid-sc-1wofa1l-0 bySYOJ"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 dDSiKk"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 ibTKOe"
                >
                  <a
                    aria-label="home"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="home"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Home
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="Calendar"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="Calendar"
                    href="/EmployeeDashboard/Schedule"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Calendar
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="MyJob"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="MyJob"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        My Job
                         
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 bqfCTq"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cdHUSp"
                    >
                       
                       
                    </span>
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Sales Associate 
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <a
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    href="/ProfilePage"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 kobaLM"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                      />
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <button
                    class="StyledButton-sc-323bzc-0 kYtlmw"
                    data-testid="logout"
                    type="button"
                  >
                    <svg
                      aria-label="Logout"
                      class="StyledIcon-sc-ofa7kd-0 cnewlt"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                        fill="none"
                        stroke="#231F20"
                        stroke-width="2"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 eETalo"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 fNigeM"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jzOfgr"
              >
                <a
                  class="StyledAnchor-sc-1rp7lwl-0 enCzim"
                  href="/EmployeeDashboard"
                >
                  My Job
                </a>
                <span
                  class="StyledText-sc-1sadyjn-0 hpCRgD"
                >
                    
                  &gt;
                    
                </span>
                <a
                  class="StyledAnchor-sc-1rp7lwl-0 enCzim"
                  href="/EmployeeDashboard/CompanyPolicy"
                >
                  Company Policies
                </a>
              </div>
              <h1>
                  Company Policy
              </h1>
              <div
                class="StyledGrid-sc-1wofa1l-0 eoFbLA"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/health.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="healthBox"
                    >
                      Health and Safety
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                  />
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/schedule.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="scheduleBox"
                    >
                      Schedule
                    </span>
                  </div>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/conduct.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="employeeBox"
                    >
                      Employee Code of Conduct
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                  />
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/opportunity.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="equalBox"
                    >
                      Equal Opportunity
                    </span>
                  </div>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/leave.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="leaveBox"
                    >
                      Leave of Absence
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                  />
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/phone.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="contactBox"
                    >
                      Contact and Address
                    </span>
                  </div>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/attendance.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="attendanceBox"
                    >
                      Attendance
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                  />
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/info.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="aboutBox"
                    >
                      About the Company
                    </span>
                  </div>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 fWaToK"
                    tabindex="0"
                  >
                    <img
                      class="StyledImage-sc-ey4zx9-0 gosVQo"
                      src="/src/assets/images/drug.png"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 bZLsdn"
                      data-testid="drugBox"
                    >
                      Substance Abuse
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
      >
        <div>
          <div
            class="StyledGrid-sc-1wofa1l-0 bySYOJ"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dDSiKk"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ibTKOe"
              >
                <a
                  aria-label="home"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="home"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Home
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="Calendar"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="Calendar"
                  href="/EmployeeDashboard/Schedule"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Calendar
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="MyJob"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="MyJob"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      My Job
                       
                    </span>
                  </div>
                </a>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 bqfCTq"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <span
                    class="StyledText-sc-1sadyjn-0 cdHUSp"
                  >
                     
                     
                  </span>
                  <span
                    class="StyledText-sc-1sadyjn-0 hpCRgD"
                  >
                     Sales Associate 
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <a
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  href="/ProfilePage"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 kobaLM"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                    />
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <button
                  class="StyledButton-sc-323bzc-0 kYtlmw"
                  data-testid="logout"
                  type="button"
                >
                  <svg
                    aria-label="Logout"
                    class="StyledIcon-sc-ofa7kd-0 cnewlt"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                      fill="none"
                      stroke="#231F20"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 eETalo"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 fNigeM"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 jzOfgr"
            >
              <a
                class="StyledAnchor-sc-1rp7lwl-0 enCzim"
                href="/EmployeeDashboard"
              >
                My Job
              </a>
              <span
                class="StyledText-sc-1sadyjn-0 hpCRgD"
              >
                  
                &gt;
                  
              </span>
              <a
                class="StyledAnchor-sc-1rp7lwl-0 enCzim"
                href="/EmployeeDashboard/CompanyPolicy"
              >
                Company Policies
              </a>
            </div>
            <h1>
                Company Policy
            </h1>
            <div
              class="StyledGrid-sc-1wofa1l-0 eoFbLA"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 iNDmPr"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/health.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="healthBox"
                  >
                    Health and Safety
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/schedule.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="scheduleBox"
                  >
                    Schedule
                  </span>
                </div>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 iNDmPr"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/conduct.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="employeeBox"
                  >
                    Employee Code of Conduct
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/opportunity.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="equalBox"
                  >
                    Equal Opportunity
                  </span>
                </div>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 iNDmPr"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/leave.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="leaveBox"
                  >
                    Leave of Absence
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/phone.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="contactBox"
                  >
                    Contact and Address
                  </span>
                </div>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 iNDmPr"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/attendance.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="attendanceBox"
                  >
                    Attendance
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/info.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="aboutBox"
                  >
                    About the Company
                  </span>
                </div>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 iNDmPr"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fWaToK"
                  tabindex="0"
                >
                  <img
                    class="StyledImage-sc-ey4zx9-0 gosVQo"
                    src="/src/assets/images/drug.png"
                  />
                  <span
                    class="StyledText-sc-1sadyjn-0 bZLsdn"
                    data-testid="drugBox"
                  >
                    Substance Abuse
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
